src/DataCollection.UWP/MainPage.xaml.cs

```diff
@@ -37,11 +37,18 @@ public MainPage()
 
             // create event handler for when receiving a message to display to the user
             UserPromptMessenger.Instance.MessageValueChanged += UserPrompt_MessageValueChanged;
+            BusyWaitingMessenger.Instance.WaitStatusChanged += OnWaitStatusChanged;
```

Comment:
Be careful here. The two lines above are hooked to a static event, which means MainPage would never be GC'ed (no unsubscribing)